### PR TITLE
Problem: export API not supported in memiavl

### PR DESCRIPTION
**Pull request by yihuang**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 - [#890](https://github.com/crypto-org-chain/cronos/pull/890) optimize memiavl snapshot format.
 - [#904](https://github.com/crypto-org-chain/cronos/pull/904) Enable "dynamic-level-bytes" on new `application.db`.
+- [#]() memiavl support `Export` API.
```
Comment (yihuang) [inline CHANGELOG.md:27]:
```suggestion
- [#924](https://github.com/crypto-org-chain/cronos/pull/924) memiavl support `Export` API.
```

File: memiavl/export.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	i        uint32
+	count    int
+}
+
+func (e *Exporter) Next() (*iavl.ExportNode, error) {
+	if int(e.i) >= e.count {
+		return nil, iavl.ExportDone
+	}
+	node := e.snapshot.Node(e.i)
+	e.i++
```
Comment (mmsqe) [inline memiavl/export.go:18]: will e.i++ dirty if concurrent?

Comment (yihuang) [reply memiavl/export.go:18]: it's not thread-safe

Comment (mmsqe) [reply memiavl/export.go:18]: so we don't need function lock?

Comment (yihuang) [reply memiavl/export.go:18]: yes, it's always [used in a sequential way](https://github.com/cosmos/cosmos-sdk/blob/main/store/rootmulti/store.go#L781), there's no point a do an iteration concurrently, the result is un-deterministic.

**Review by mmsqe — APPROVED** [review]

**Pull request merged**